### ISSUE TITLE
Transformed field raises KeyError when data type not specified

### ISSUE THREAD

**Issue by malramsay64** [altair-viz/altair#587]
When using a transform and missing the type specification, Altair raises a KeyError which isn't handled properly raising an additional KeyError. This can be replicated using the code below, which is missing the type on the X axis.

```
import altair as alt

cars = alt.load_dataset('cars')

chart = alt.Chart(cars).mark_point().encode(
    x=alt.X('watts', axis=alt.Axis(title='W'), bin=alt.Bin(maxbins=15)),
    y=alt.Y('count(*):Q', axis=alt.Axis(title='# Cars')),
).transform_calculate(
   'watts', 'datum.Horsepower * 746' 
)
chart
```
When the X axis has the type and it is instead removed from the Y axis, a SchemaValidationError is raised specifying `'type' is a required property` which I would expect to be the intended behaviour in the above example.

**Comment by jakevdp** [comment]
Thanks!

This has to do with the type inference code... when a column is missing from the dataframe, it is not handled gracefully.

I'll take a look, probably tomorrow :smile:

**Comment by jakevdp** [comment]
Fixed in 3947cddd0fcd2bede2289ba491566b4ed87955ab
